pjlib/src/pj/ioqueue_winnt.c

```diff
@@ -698,6 +715,49 @@ static void decrement_counter(pj_ioqueue_key_t *key)
 }
 #endif
 
+static struct pending_op *alloc_pending_op(pj_ioqueue_key_t *key, pj_ioqueue_op_key_t *op_key)
+{
+    pj_pool_t *pool;
+    struct pending_op *op;
+
+    pool = pj_pool_create(key->ioqueue->pool->factory, "pending_op%p", 512, 512, NULL);
+    PJ_ASSERT_RETURN(pool, NULL);
+    op = PJ_POOL_ZALLOC_T(pool, struct pending_op);
+    op->pool = pool;
+    op->op_key = op_key;
+
+    pj_mutex_lock(key->mutex);
+    pj_list_push_back(&key->pending_list, op);
+    pj_mutex_unlock(key->mutex);
+    return op;
+}
+
+static void release_pending_op(pj_ioqueue_key_t *key, struct pending_op *op)
+{
+    if (!key || !op)
+        return;
+    pj_mutex_lock(key->mutex);
+    pj_list_erase(op);
+    pj_mutex_unlock(key->mutex);
+    pj_pool_release(op->pool);
+}
+
+static void cancel_all_pending_op(pj_ioqueue_key_t *key)
+{
+    if (!fnCancelIoEx)
+        fnCancelIoEx = (FnCancelIoEx)GetProcAddress(GetModuleHandle(PJ_T("Kernel32.dll")), "CancelIoEx");
+    if (fnCancelIoEx) {
+        struct pending_op *op;
+        pj_mutex_lock(key->mutex);
+        for (op = key->pending_list.next; op != &key->pending_list; op = op->next) {
+            BOOL rc = fnCancelIoEx(key->ioqueue->iocp, (LPOVERLAPPED)&op->pending_key);
+            if (rc)
+                PJ_PERROR(2, (THIS_FILE, PJ_RETURN_OS_ERROR(GetLastError()), "cancel io error"));
```

Comment:
- actually when `rc ==1`, it succeeds
- shouldn't the handle `key->hnd` instead of `key->ioqueue->iocp`?